### PR TITLE
API tidy up, comments, remove unused code etc

### PR DESCRIPTION
**Pull request by mpickering**

### CODE REVIEW

File: hie-bios.cabal [hunk 1]
```diff
@@ -27,17 +27,16 @@ Library
                         HIE.Bios.Config
                         HIE.Bios.Cradle
                         HIE.Bios.Environment
-                        HIE.Bios.Debug
+                        HIE.Bios.Internal.Debug
```
Comment (fendor) [inline hie-bios.cabal]: Listed modules were sorted. If it doesnt matter, not a blocker, just sayin'

File: src/HIE/Bios/Ghc/Logger.hs [hunk 2]
```diff
@@ -35,11 +31,11 @@ newtype LogRef = LogRef (IORef Builder)
...
 newLogRef = LogRef <$> newIORef id
 
-readAndClearLogRef :: Options -> LogRef -> IO String
-readAndClearLogRef opt (LogRef ref) = do
+readAndClearLogRef :: LogRef -> IO String
+readAndClearLogRef (LogRef ref) = do
     b <- readIORef ref
     writeIORef ref id
-    return $! convert opt (b [])
+    return $! unlines (b [])
```
Comment (fendor) [inline src/HIE/Bios/Ghc/Logger.hs]: Out of interest, I think this is one of the very few places where `($!)` is used. Is there a reason for it?

**Comment by fendor** [comment]
Haddock Documentation is a bit broken. We can fix that in a follow-up PR, though.

File: src/HIE/Bios/Ghc/Logger.hs (hunk 2 above)
Comment (mpickering) [reply src/HIE/Bios/Ghc/Logger.hs]: This is old code from `hpp` still so I can't comment on the reason.

**Pull request merged**